src/ui/common/src/components/tables/WorkflowTable.tsx

```diff
@@ -0,0 +1,252 @@
+import { faSearch, faX } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import Box from '@mui/material/Box';
+import Paper from '@mui/material/Paper';
+import Table from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import TableCell from '@mui/material/TableCell';
+import TableContainer from '@mui/material/TableContainer';
+import TableHead from '@mui/material/TableHead';
+import TablePagination from '@mui/material/TablePagination';
+import TableRow from '@mui/material/TableRow';
+import TextField from '@mui/material/TextField';
+import Typography from '@mui/material/Typography';
+import * as React from 'react';
+
+import { theme } from '../../styles/theme/theme';
+import { DataSchema } from '../../utils/data';
+import { CheckPreview } from '../pages/workflows/components/CheckItem';
+import { ExecutionStatusLinkProps } from '../pages/workflows/components/ExecutionStatusLink';
+import { MetricPreview } from '../pages/workflows/components/MetricItem';
+
+export type WorkflowTableElement = string | number | boolean | JSX.Element;
+
+export type WorkflowTableRow = {
+  [key: string]: WorkflowTableElement;
+};
+
+export type WorkflowTableRowData = {
+  [key: string]:
+    | string
+    | number
+    | boolean
+    | CheckPreview[]
+    | MetricPreview[]
+    | ExecutionStatusLinkProps;
+};
+
+export interface WorkflowTableData {
+  schema?: DataSchema;
+  data: WorkflowTableRow[];
+  meta: WorkflowTableRowData[];
+}
+
+export interface WorkflowsTableProps {
+  data: WorkflowTableData;
+  searchEnabled?: boolean;
+  onGetColumnValue?: (row, column) => WorkflowTableElement;
+}
+
+export const WorkflowTable: React.FC<WorkflowsTableProps> = ({
+  data,
+  onGetColumnValue,
+  searchEnabled = false,
+}) => {
+  const [page, setPage] = React.useState(0);
+  const [rowsPerPage, setRowsPerPage] = React.useState(5);
+  const [searchQuery, setSearchQuery] = React.useState('');
+  // TODO: Add dropdown to select which column to search the table on.
+  const [searchColumn, setSearchColumn] = React.useState('name');
+
+  let rows = data.data;
+  const columns = data.schema.fields;
+
+  let filteredRows = [];
+
+  const shouldInclude = (rowItem, searchQuery, searchColumn): boolean => {
```

Comment:
Update this so that we check if a prop is passed in called shouldInclude, and call that function instead.

Include some documentation on how to to use this function.
(a notion doc / storybook documentation is a good idea here)